libs/base/ui/overlays/notification-center/src/registry.controller.ts

```diff
@@ -1,142 +1,110 @@
     }
-
-    this.toggleVisibility(targetKey);
+    this.items.unshift(registryItem);
+    this.host.requestUpdate();
+    setTimeout(() => {
+      registryItem.visible = true;
+      this.host.requestUpdate();
+    }, 0);
   }
 
-  handleNotificationClose(key: string): void {
-    this.removeFromRegistry(key, true);
+  protected scheduleAutoClosing(item: NotificationRegistry): void {
+    // const key = item.key as string;
+    this.autoCloseQueue[item.key] = {
+      autoCloseTime: item.autoCloseTime,
+      timeout: setTimeout(() => {
+        this.destroy(item.key);
+      }, item.autoCloseTime),
+    };
   }
 
-  handleTransitionEnd(key: string): void {
-    if (!this.notificationIsVisible(key)) {
-      this.removeFromRegistry(key);
+  protected resolveStrategy(item: Notification): Notification {
+    const strategy = { floating: true, closable: true, ...item };
+
+    if (strategy.autoClose === undefined) {
+      strategy.autoClose =
+        strategy.type === AlertType.Info || AlertType.Success === strategy.type;
     }
-  }
 
-  protected removeFromRegistry(targetKey: string, force = false): void {
-    if (force) {
-      //should clear the delayed callback
-      this.preventAutoClose(targetKey);
+    if (!strategy.closable && !strategy.autoClose) {
+      strategy.closable = true;
     }
-    this.registry = this.registry.filter(({ key }) => targetKey !== key);
+    return strategy;
   }
 
-  protected toggleVisibility(targetKey: string): void {
-    this.registry = this.registry.map((item) => {
-      if (item.key === targetKey) {
-        return {
-          ...item,
-          visible: !item.visible,
-          _mounted: true,
-        };
-      }
-
-      return item;
-    });
+  constructor(public host: LitElement) {
+    this.host.addController(this);
   }
 
-  protected notificationIsVisible(targetKey: string): boolean {
-    return Boolean(
-      (
-        this.registry.find(
-          ({ key }) => key === targetKey
-        ) as NotificationRegistry
-      )?.visible
+  hostConnected?(): void {
+    this.host.addEventListener('oryx.close', (ev: any) =>
+      this.onClose(ev.detail.key)
     );
+    this.host.addEventListener('mouseenter', () => this.onMouseEnter());
+    this.host.addEventListener('mouseleave', () => this.continueAutoClose());
   }
 
-  //need to set visible attr to true for new notifications to run animation
-  protected ensureNotificationMountedState(): void {
-    if (this.registry.some(({ key, _mounted }) => key && !_mounted)) {
-      setTimeout(() => {
-        this.registry = this.registry.map((strategy) => ({
-          ...strategy,
-          visible: true,
-          _mounted: true,
-        }));
-      }, 0);
-    }
+  protected onClose(key: number): void {
+    this.destroy(key);
   }
 
-  constructor(public host: LitElement) {
-    this.host.addController(this);
+  protected onMouseEnter(): void {
+    Object.keys(this.autoCloseQueue).forEach((key) => {
```

Comment:
Should it prevent auto close of all notifications or thats one that was mousevered?

Reply:
all